Engine mutable receiver

aludvik opened the pull request.

Commit:
Take mutable receiver in Engine.start() method

Previously, this took an immutable receiver because of the requirement that
engine be Sync, but not that this has been lifted, this can (and should) take a
mutable receiver.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

Commit:
Execute Engine.start in calling thread

Instead of moving the Engine to a background thread, run it in the calling
thread. This eliminates many restrictions on how Engine is implemented. A few
driver API specifics need to be tweaked in order to allow the calling function
to stop the driver.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

Commit:
Remove Send trait bound from Engine

By running the Engine in the calling thread, the Send trait bound can be
removed.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

nick-drozd commented on sdk/rust/src/consensus/zmq_driver.rs:
```diff
@@ -74,13 +74,14 @@ pub fn start<T: AsRef<str>, E: Engine + Send + 'static >(endpoint: T, engine: E)
                 Box::new(ZmqService::new(
                     validator_sender_clone,
                     ::std::time::Duration::from_secs(SERVICE_TIMEOUT),
-                    engine.name(),
```
Is this a functional change?

aludvik replied:
In order to borrow the engine as mutable here, it needs to not be borrowed anywhere else.

Commit:
FIXUP: Delete driver trait

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

nick-drozd commented:
tested locally, works

Commit:
Add use statements for imports from std library

Import items into the local namespace instead of using fully-qualified names.
This is more consistent with the style used in the rest of Sawtooth.

Signed-off-by: Adam Ludvik <ludvik@bitwise.io>

nick-drozd commented:
works
